tt_metal/jit_build/hlk_desc.hpp

```diff
@@ -29,34 +30,24 @@ class tt_hlk_desc
     size_t hlk_args_size;  // size of hlk_args_t in bytes (result of sizeof())
 
    public:
-    DataFormat input_buf_dataformat_arr[8];
-    DataFormat param_buf_dataformat_arr[8];
-    DataFormat output_buf_dataformat_arr[8];
-    DataFormat intermediate_buf_dataformat_arr[8];
-    uint32_t buf_num_faces_arr[32];
-    uint32_t buf_partial_face_arr[32];
-    uint32_t buf_face_r_dim_arr[32];
-    uint32_t buf_narrow_tile_arr[32];
-    uint32_t buf_tile_r_dim_arr[32];
-    uint32_t buf_tile_c_dim_arr[32];
-    uint32_t buf_tile_size_arr[32];
+    DataFormat buf_dataformat_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_num_faces_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_partial_face_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_face_r_dim_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_narrow_tile_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_tile_r_dim_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_tile_c_dim_arr[NUM_CIRCULAR_BUFFERS];
+    uint32_t buf_tile_size_arr[NUM_CIRCULAR_BUFFERS];
 
     tt_hlk_desc() {
         math_fidelity = MathFidelity::Invalid;
         hlk_args = nullptr;
         hlk_args_size = 0;
         approximation_mode = true;
 
-        for (int i = 0; i < 8; ++i)
-        {
-            input_buf_dataformat_arr[i] = DataFormat::Invalid;
-            param_buf_dataformat_arr[i] = DataFormat::Invalid;
-            output_buf_dataformat_arr[i] = DataFormat::Invalid;
-            intermediate_buf_dataformat_arr[i] = DataFormat::Invalid;
-        }
-
-        for (int i = 0; i < 32; ++i)
+        for (int i = 0; i < NUM_CIRCULAR_BUFFERS; ++i)
```

Comment:
Just curious of uint32_t vs std::array usage. Would you go for an array if this was implemented from scratch?

Reply:
Probably just one std::array with an aggregate type of all these variables.